x-pack/plugins/saved_objects_tagging/public/plugin.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { i18n } from '@kbn/i18n';
+import { CoreSetup, CoreStart, PluginInitializerContext, Plugin } from 'src/core/public';
+import { ManagementSetup } from '../../../../src/plugins/management/public';
+import { SavedObjectTaggingOssPluginSetup } from '../../../../src/plugins/saved_objects_tagging_oss/public';
+import { tagManagementSectionId } from '../common/constants';
+import { getTagsCapabilities } from '../common/capabilities';
+import { SavedObjectTaggingPluginStart } from './types';
+import { TagsClient, TagsCache } from './tags';
+import { getUiApi } from './ui_api';
+import { SavedObjectsTaggingClientConfig, SavedObjectsTaggingClientConfigRawType } from './config';
+
+interface SetupDeps {
+  management: ManagementSetup;
+  savedObjectsTaggingOss: SavedObjectTaggingOssPluginSetup;
+}
+
+export class SavedObjectTaggingPlugin
+  implements Plugin<{}, SavedObjectTaggingPluginStart, SetupDeps, {}> {
+  private tagClient?: TagsClient;
+  private tagCache?: TagsCache;
+  private readonly config: SavedObjectsTaggingClientConfig;
+
+  constructor(context: PluginInitializerContext) {
+    this.config = new SavedObjectsTaggingClientConfig(
+      context.config.get<SavedObjectsTaggingClientConfigRawType>()
+    );
+  }
+
+  public setup(
+    core: CoreSetup<{}, SavedObjectTaggingPluginStart>,
+    { management, savedObjectsTaggingOss }: SetupDeps
+  ) {
+    const kibanaSection = management.sections.section.kibana;
+    kibanaSection.registerApp({
+      id: tagManagementSectionId,
+      title: i18n.translate('xpack.savedObjectsTagging.management.sectionLabel', {
+        defaultMessage: 'Tags',
+      }),
+      order: 2,
+      mount: async (mountParams) => {
+        const { mountSection } = await import('./management');
+        return mountSection({
+          tagClient: this.tagClient!,
+          core,
+          mountParams,
+        });
+      },
+    });
+
+    savedObjectsTaggingOss.registerTaggingApi(
+      core.getStartServices().then(([_core, _deps, startContract]) => startContract)
```

Comment:
who said async registry?😅

Reply:
Yet another use case of start contract access requirement during setup. I decided to try the 'async register handler' approach this time.